TF-M: Remove reserved memory for FLPR for Nordic nRF54L

Vge0rge opened the pull request:
Update the memory layout of the Nordic nRF54L devices to avoid reserving volatile and non-volatile memory for FLPR since it is not yet supported with TF-M.

This is done for nRF54L10/nRF54L15/nRF54LM20A devices.

In order to do that some restructuring was done to the dts files when possible.
The RAM and RRAM partitions are now all defined in the existed  ns_partition.dtsi file so that it can be reused from all boards.

Vge0rge committed:
dts: nordic: nrf54l15 ns refactor/update memory mapping

Update the memory layout of nrf54l15 for TF-M builds so that
it does not reserve any memory for FLPR since it is not supported
with TF-M.

This affects both the SRAM and the RRAM partitioning.

To do that I refactored the dts files, specifically:
1) I created new files for the _ns targets since they have
   different available RRAM/SRAM sizes now.
2) I moved the SRAM partitioning to the nrf54l15_ns_partition.dtsi
   and removed it from individual board files so it can be updated
   for all the platforms in one place.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
dts: nordic: nrf54lm20a ns refactor/update memory mapping

Update the memory layout of nrf54lm20a for TF-M builds so that
it does not reserve any memory for FLPR since it is not supported
with TF-M.

This affects both the SRAM and the RRAM partitioning.

I moved the SRAM partitioning to the nrf54lm20a_ns_partition.dtsi
and removed it from individual board files so it can be updated
for all the platforms in one place.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
dts: nordic: nrf54l10 ns refactor/update memory mapping

Update the memory layout of nrf54l10 for TF-M builds so that
it does not reserve any memory for FLPR since it is not supported
with TF-M.

This affects both the SRAM and the RRAM partitioning.

I moved the SRAM partitioning to the nrf54l10_ns_partition.dtsi
and removed it from individual board files so it can be updated
for all the platforms in one place.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge committed:
manifest: tf-m: Remove reserved memory for FLPR in TF-M for Nordic

Update the memory layout of the Nordic nRF54L devices to avoid
reserving volatile and non-volatile memory for FLPR since it is
not yet supported with TF-M.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

Vge0rge commented on boards/ezurio/bl54l15_dvk/bl54l15_dvk_nrf54l15_cpuapp_ns.dts:
```diff
@@ -9,7 +9,7 @@
 #define USE_NON_SECURE_ADDRESS_MAP 1
 
-#include <nordic/nrf54l15_cpuapp.dtsi>
+#include <nordic/nrf54l15_cpuappns.dtsi>
```
this is how I initially did it but I saw that the 5340 has it as cpuappns and I wanted the two to have the same naming. But I can rename both if its a big deal.

Vge0rge commented on dts/vendor/nordic/nrf54l15_ns_partition.dtsi:
```diff
@@ -4,6 +4,34 @@
+		#size-cells = <1>;
+		ranges;
+
+		sram0_s: image_s@20000000 {
```
I am not sure that I follow sorry, these are reserved areas since they are under the reserve area dts entry no?

Also this is a copy of what existed already in the cpuapp board files, I didn't change anything apart from the ranges/sizes here. What are you suggesting exactly?

Vge0rge commented on dts/vendor/nordic/nrf54l15_ns.dtsi:
```diff
@@ -0,0 +1,16 @@
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
You are commenting in the wrong file, now I see what happened. I will update the partition because there indeed I forgot the cpuapp, this file will stay as is.

Vge0rge replied:
There are two files:
./dts/arm/nordic/nrf54l15_cpuappns.dtsi
./dts/vendor/nordic/nrf54l15_ns.dtsi

I cannot update both to be nrf54l15_cpuapp_ns.dtsi, that does't make sense to me at least.

I will update the nrf54l15_cpuappns to be nrf54l15_cpuapp_ns as you requested fro the 54l15 and the 5340. Lets fix the missing cpuapp in another PR where more restructuring can happen.

Vge0rge committed:
dts: nordic: nrf54l refactor SRAM partitioning for NS builds

Refactor the SRAM partitioning for TF-M builds for the Nordic
nRF54L devices. Instead of using the reserved-memory node this
just partitions the normal SRAM node. This aligns the design with
the rest of the Nordic devices.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

tomi-font commented:
> I fixed the compliance, I am not sure what is the Issue Assigner doing though.

@nashif @stephanosio -- @Vge0rge can you also try rebasing your PR? There have been recent changes to the script.

tomi-font commented on dts/vendor/nordic/nrf54l15_ns_partition.dtsi:
```diff
@@ -51,12 +78,12 @@
+		storage_partition: partition@175000 {
 			label = "storage";
-			reg = <0x00015D000 DT_SIZE_K(32)>;
+			reg = <0x000175000 DT_SIZE_K(32)>;
```
Will this partitioning change break compatibility when upgrading from an older release?

Vge0rge replied:
For sure it will! But since we don't support any bootloaders upstream and thus no DFU is possible this is a non issue :)

tomi-font commented:
Seems a rebase won't work. See https://github.com/zephyrproject-rtos/zephyr/pull/99726.

Vge0rge committed:
dts: nordic: Rename all Nordic dtsi files for non secure

Rename all dtsi files for Nordic boards for non secure builds
to have a _ prefix before "ns" to increase readiblity.

At the same time, change the path to include the arm
folder in order to be able to differentiate with the
vendor folder which has similar files.

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

kylebonnici commented on dts/vendor/nordic/nrf54lm20a.dtsi:
```diff
@@ -99,12 +99,25 @@
 #endif
 
+#ifdef USE_NON_SECURE_ADDRESS_MAP
+
```
No not yet see https://github.com/zephyrproject-rtos/zephyr/pull/100247

But if this PR is merged as is then https://github.com/zephyrproject-rtos/zephyr/pull/100247 can fix this after.

Vge0rge committed:
dts: nordic: Rename the dts/vendor dtsi files

This renames the dtsi files in the dts/vendor/nordic
to include the cpuapp in the filename.

At the same time it updates the inclusion of these files
to include the vendor in the path because the same filenames
exists in both:
dts/vendor/nordic
dts/arm/nordic

Signed-off-by: Georgios Vasilakis <georgios.vasilakis@nordicsemi.no>

nordicjm commented:
Rebase and address compliance issues first. Fix the issues in the original commits without adding them then reworking them in a fixup commit. And there are outstanding comments e.g. not to use reserved memory for application SRAM usage and put it in the soc node. Note: use sub-partitions (child nodes) which are part of the main SRAM node to partition it down, not separate nodes altogether

Vge0rge commented:
> Rebase and address compliance issues first. Fix the issues in the original commits without adding them then reworking them in a fixup commit. And there are outstanding comments e.g. not to use reserved memory for application SRAM usage and put it in the soc node. Note: use sub-partitions (child nodes) which are part of the main SRAM node to partition it down, not separate nodes altogether

Well, I am not willing to spend more time on this. This PR was about just removing a single partition. But then you requested to rename the whole dtsi/dtsi file and change the approach that we did the inclusion.  Now you request to change the fixup commits and to rework the whole PR and also change the approach of the reserved partitions. 

I do agree the reserved memory entries need to get fixed at some point but I don't agree with the approach that this PR needs to fix everything that is wrong just because it made it visible to you. It is already out of scope what I am doing here. We have tasks and we can track issues to be fixed later.

So basically if you find functional bug fixes then I am happy to update this PR. (Fixing the fixup commits I don't consider a bug fix). Otherwise the choice is to either take over the PR or I am gonna close it as I said before.

Vge0rge commented:
> > Well, I am not willing to spend more time on this. This PR was about just removing a single partition. But then you requested to rename the whole dtsi/dtsi file and change the approach that we did the inclusion. Now you request to change the fixup commits and to rework the whole PR and also change the approach of the reserved partitions.
> 
> It's literally in the contributor guidlines https://docs.zephyrproject.org/latest/contribute/contributor_expectations.html#pr-requirements
> 
> > No fixup or merge commits are allowed, see [Contribution Workflow](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow) for more information.
> 
> I'm not asking you to do it because I'm bored?
> 

Of course it is but these commits are not made with "git commit --fixup" as the code guidelines suggest as I understand them. They are just subsequent improvements that actually have different scope compared to the first commits.

> > I do agree the reserved memory entries need to get fixed at some point but I don't agree with the approach that this PR needs to fix everything that is wrong just because it made it visible to you. It is already out of scope what I am doing here. We have tasks and we can track issues to be fixed later.
> 
> What you have here is not valid with the DT spec, @kylebonnici can give you further information I will also point out I offered to actually take this PR off your hands and fix it, and you declined. I'm still willing to fix the PR up if you don't want to

You did offer to help indeed and at the time I thought that there is no need for you to spend more time on this. But now I am happy  to accept help, please take this over and change it as you wish. I will fix the compliance issues a bit later today though, don't start working on it before this is fixed.
